README
NAME
    Dist::Zilla::PluginBundle::Author::DOHERTY - configure Dist::Zilla like
    DOHERTY

VERSION
    version 0.017

SYNOPSIS
        # in dist.ini
        [@Author::DOHERTY]

DESCRIPTION
    "Dist::Zilla::PluginBundle::Author::DOHERTY" provides shorthand for a
    Dist::Zilla configuration approximately like:

        [Git::Check]
        [@Filter]
        -bundle = @Basic    ; Equivalent to using [@Basic]
        -remove = Readme    ; For use with [CopyReadmeFromBuild]
        -remove = ExtraTests

        [AutoPrereqs]
        [MinimumPerl]
        [Github::Meta]
        [PodWeaver]
        config_plugin = @Author::DOHERTY
        [InstallGuide]
        [ReadmeFromPod]
        [CopyReadmeFromBuild]
        [CopyMakefilePLFromBuild]

        [Git::NextVersion]
        [PkgVersion]
        [NextRelease]
        filename = Changes
        format   = %-9v %{yyyy-MM-dd}d
        [CheckChangesHasContent]
        changelog = Changes

        [Twitter]         ; config in ~/.netrc
        [Github::Update]  ; config in ~/.gitconfig
        [Git::Commit]
        [Git::Tag]

        [@TestingMania]
        changelog = Changes
        [LocalInstall]

USAGE
    Just put "[@Author::DOHERTY]" in your dist.ini. You can supply the
    following options:

    *   "fake_release" specifies whether to use "FakeRelease" instead of
        "UploadToCPAN". Defaults to 0.

    *   "add_tests" is a comma-separated list of testing plugins to add to
        "TestingMania".

    *   "skip_tests" is a comma-separated list of testing plugins to skip in
        "TestingMania".

    *   "tag_format" specifies how a git release tag should be named. This
        is passed to "Git::Tag".

    *   "version_regex" specifies a regexp to find the version number part
        of a git release tag. This is passed to "Git::NextVersion".

    *   "no_twitter" says that releases of this module shouldn't be tweeted.

    *   "surgical" says to use Dist::Zilla::Plugin::SurgicalPodWeaver.

    *   "changelog" is the filename of the changelog, and defaults to
        Changes.

SEE ALSO
    "Dist::Zilla"

AVAILABILITY
    The project homepage is
    <http://p3rl.org/Dist::Zilla::PluginBundle::Author::DOHERTY>.

    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Dist-Zilla-PluginBundle-Author-DOHERTY/>.

    The development version lives at
    <http://github.com/doherty/Dist-Zilla-PluginBundle-Author-DOHERTY> and
    may be cloned from
    <git://github.com/doherty/Dist-Zilla-PluginBundle-Author-DOHERTY.git>.
    Instead of sending patches, please fork this project using the standard
    git and github infrastructure.

SOURCE
    The development version is on github at
    <http://github.com/doherty/Dist-Zilla-PluginBundle-Author-DOHERTY> and
    may be cloned from
    <git://github.com/doherty/Dist-Zilla-PluginBundle-Author-DOHERTY.git>

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <https://github.com/doherty/Dist-Zilla-PluginBundle-Author-DOHERTY/issue
    s>.

AUTHOR
    Mike Doherty <doherty@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Mike Doherty.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



CHANGES
Revision history for Perl module Dist-Zilla-PluginBundle-Author-DOHERTY

{{$NEXT}}
    * Update Github plugins
    * Remove TestingMania version prerequisite

0.015     2011-04-02
    * Add support for specifying the changelog filename

0.014     2011-02-18
    * Attempt to use Git::Commit and Git::Push properly
    * Use OurPkgVersion instead of PkgVersion - much better
    * Require 0.0017 for ROKR's plugins - bugfixes

0.012     2011-02-06
    * Require D::Z::PB::TestingMania 0.003 for a better dep tree & PodLinkTests
    * Add CopyMakefilePLFromBuild

0.011     2011-01-24
    * Use new name for Pod::Weaver pluginbundle

0.010     2011-01-24
    * Renamed to Dist::Zilla::PluginBundle::Author::DOHERTY
    * Added support for using SurgicalPodWeaver via surgical in dist.ini
    * Bumped minimum version of InstallRelease to hopefully detect failed installs

0.009     2011-01-23
    * Handle the case where it is a fake release
    * Use is.gd to shorten URL for Twitter

0.008     2011-01-18
    * Fixed spurious warning
    * Require support for p3rl.org in GithubUpdate

0.007     2011-01-18
    * Fixed undef plugin error when no_twitter is set
    * Allow README to be dirty when releasing (again?)

0.006     2011-01-18
    * Tweet releases
    * Update github information on release
    * Relicensed under the same terms as Perl 5

0.005     2011-01-11
    * Re-upload properly - no changes since 0.004

0.004     2011-01-11
    * Avoid circular depenencies in Dist::Zilla::PluginBundle::ROKR (which provides CopyReadmeFromBuild)
    * Add Dist::Zilla::Plugin::CheckExtraTests dependency, which provides the xtest command in addition to the plugin itself
    * Fix Git::Commit so it really commits
    * Bumped version of Pod::Weaver::PluginBundle::DOHERTY

0.003     2010-11-24
    * Require Pod::Weaver::PluginBundle::DOHERTY
    * Allow README to be dirty - it always will be since we copy from build

0.002     2010-11-13
    * Fix passing skip_tests and add_tests to Dist::Zilla::PluginBundle::TestingMania

0.001     2010-10-31
    * Initial version


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.006; }

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'configure Dist::Zilla like DOHERTY',
  'AUTHOR' => 'Mike Doherty <doherty@cpan.org>',
  'BUILD_REQUIRES' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'Test::More' => '0'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Dist-Zilla-PluginBundle-Author-DOHERTY',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Dist::Zilla::PluginBundle::Author::DOHERTY',
  'PREREQ_PM' => {
    'Dist::Zilla' => '4.102341',
    'Dist::Zilla::Plugin::CheckChangesHasContent' => '0',
    'Dist::Zilla::Plugin::CheckExtraTests' => '0',
    'Dist::Zilla::Plugin::CopyMakefilePLFromBuild' => '0.0017',
    'Dist::Zilla::Plugin::CopyReadmeFromBuild' => '0.0017',
    'Dist::Zilla::Plugin::Git::Check' => '0',
    'Dist::Zilla::Plugin::Git::Commit' => '0',
    'Dist::Zilla::Plugin::Git::NextVersion' => '0',
    'Dist::Zilla::Plugin::Git::Tag' => '0',
    'Dist::Zilla::Plugin::GitHub::Meta' => '0.06',
    'Dist::Zilla::Plugin::GitHub::Update' => '0.06',
    'Dist::Zilla::Plugin::InstallGuide' => '0',
    'Dist::Zilla::Plugin::InstallRelease' => '0.006',
    'Dist::Zilla::Plugin::MinimumPerl' => '0',
    'Dist::Zilla::Plugin::OurPkgVersion' => '0',
    'Dist::Zilla::Plugin::PodWeaver' => '0',
    'Dist::Zilla::Plugin::ReadmeFromPod' => '0',
    'Dist::Zilla::Plugin::SurgicalPodWeaver' => '0.0015',
    'Dist::Zilla::Plugin::Twitter' => '0.010',
    'Dist::Zilla::PluginBundle::TestingMania' => '0.009',
    'Dist::Zilla::Role::PluginBundle::Easy' => '0',
    'Moose' => '0.99',
    'Moose::Autobox' => '0',
    'Pod::Weaver::PluginBundle::Author::DOHERTY' => '0.004',
    'Pod::Weaver::Section::BugsAndLimitations' => '1.102670',
    'WWW::Shorten::Googl' => '0',
    'WWW::Shorten::IsGd' => '0',
    'autodie' => '2.00',
    'namespace::autoclean' => '0.09'
  },
  'VERSION' => '0.017',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



